src/Package/Impl/Utilities/ProjectUtilities.cs

```diff
@@ -56,10 +56,20 @@ public static class ProjectUtilities {
         public static EnvDTE.Project GetActiveProject() {
             DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
             if (dte.Solution.Projects.Count > 0) {
-                var projects = (Array)dte.ActiveSolutionProjects;
-                if (projects != null && projects.Length == 1) {
-                    return (EnvDTE.Project)projects.GetValue(0);
-                }
+                try {
+                    if (dte.Solution != null) {
+                        Array projects;
+                        var solutionFile = dte.Solution.FullName;
+                        if (!string.IsNullOrEmpty(solutionFile)) {
+                            projects = (Array)dte.ActiveSolutionProjects;
+                            return (EnvDTE.Project)projects.GetValue(0);
+                        } else {
+                            var e = dte.Solution.Projects.GetEnumerator();
```

Comment:
Enumerator needs to be disposed.

Better yet, you can do `Projects.Cast<EnvDTE.Project>().First()`, to avoid having to muck around with enumerators directly.